tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/CalendarEventCounterAcceptMethodContract.scala

```diff
@@ -0,0 +1,744 @@
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .body("methodResponses[0][1].accepted", contains(counterEventBlobId))
+
+    // Then: The calendar should be updated on the DAV server with the new proposed dates
+    val parsedCalendar: CalendarEventParsed = CalendarEventParsed.from(getCalendarFromDav(bobCredential, eventUid)).head
+    assertThat(parsedCalendar.startAsJava().get().truncatedTo(ChronoUnit.SECONDS)).isEqualTo(counterProposedStartDate.truncatedTo(ChronoUnit.SECONDS))
+    assertThat(parsedCalendar.endAsJava().get().truncatedTo(ChronoUnit.SECONDS)).isEqualTo(counterProposedEndDate.truncatedTo(ChronoUnit.SECONDS))
+  }
+
+  @Test
+  def shouldReturnNotAcceptedWhenEventMissingOnDavServer(server: GuiceJamesServer): Unit = {
+    // Given: An event that does not exist on the DAV server
+    val eventUid: String = UUID.randomUUID().toString
+    val originalCalendar: CalendarEventHelper = CalendarEventHelper(
+      uid = eventUid,
+      start = ZonedDateTime.now(),
+      end = ZonedDateTime.now().plusHours(1),
+      attendee = aliceCredential.username.asString(),
+      organizer = Some(bobCredential.username.asString()))
+
+    // And: A counter event proposing a new schedule
+    val counterEvent: Calendar = originalCalendar.generateCounterEvent(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusHours(1).minusDays(1))
+    val counterEventBlobId: String = createNewEmailWithCalendarAttachment(server, eventUid, counterEvent)
+
+    // When: The counter event is accepted
+    val response: String =
+      `given`
+        .body(
+          s"""{
+             |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:calendar:event"],
+             |  "methodCalls": [[
+             |    "CalendarEventCounter/accept",
+             |    {
+             |      "accountId": "$bobAccountId",
+             |      "blobIds": [ "$counterEventBlobId" ]
+             |    },
+             |    "c1"]]
+             |}""".stripMargin)
+      .when
+        .post
+      .`then`
+        .statusCode(SC_OK)
+        .contentType(JSON)
+        .extract
+        .body
+        .asString
+
+    // Then: The response should indicate that the event was not accepted due to missing on the DAV server
+    assertThatJson(response)
+      .withOptions(IGNORING_ARRAY_ORDER)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |  "CalendarEventCounter/accept",
+           |  {
+           |    "accountId": "$bobAccountId",
+           |    "accepted": [],
+           |    "notAccepted": {
+           |      "$counterEventBlobId": {
+           |        "type": "serverFail",
+           |        "description": "$${json-unit.ignore}"
+           |      }
+           |    }
+           |  },
+           |  "c1"
+           |]""".stripMargin)
+  }
+
+
+  @Test
+  def shouldReturnNotFoundWhenBlobIdDoesNotExists() : Unit = {
+    val counterBlobId: String = new Random().nextInt(1_000_000) + ""
```

Comment:
That's memory only.

Please hide this behing and interface method as per beest practices.